lib/cli/config.js

```diff
@@ -24,30 +24,45 @@ const utils = require('../utils');
 exports.CONFIG_FILES = [
   '.mocharc.cjs',
   '.mocharc.js',
+  '.mocharc.mjs',
   '.mocharc.yaml',
   '.mocharc.yml',
   '.mocharc.jsonc',
   '.mocharc.json'
 ];
 
+/**
+ * Loads a CommonJS or ESM module, resolving its path if the file path is not
+ * relative
+ *
+ * @param {string} filepath - Module file path to load
+ * @returns {Object} CommonJS or ESM module
+ */
+const loadModule = filepath => {
+  let cwdFilepath;
+  try {
+    debug('parsers: load cwd-relative path: "%s"', path.resolve(filepath));
+    cwdFilepath = require.resolve(path.resolve(filepath)); // evtl. throws
+    return require(cwdFilepath);
+  } catch (err) {
+    if (cwdFilepath) throw err;
+
+    debug('parsers: retry load as module-relative path: "%s"', filepath);
+    return require(filepath);
+  }
+};
+
 /**
  * Parsers for various config filetypes. Each accepts a filepath and
  * returns an object (but could throw)
  */
 const parsers = (exports.parsers = {
   yaml: filepath => require('js-yaml').load(fs.readFileSync(filepath, 'utf8')),
-  js: filepath => {
-    let cwdFilepath;
-    try {
-      debug('parsers: load cwd-relative path: "%s"', path.resolve(filepath));
-      cwdFilepath = require.resolve(path.resolve(filepath)); // evtl. throws
-      return require(cwdFilepath);
-    } catch (err) {
-      if (cwdFilepath) throw err;
+  js: loadModule,
+  mjs: filepath => {
+    const module = loadModule(filepath);
 
-      debug('parsers: retry load as module-relative path: "%s"', filepath);
-      return require(filepath);
-    }
+    return module.default ?? module;
```

Comment:
+1 to me on the `module.default ?? module` change. I think it might make sense to do it for `js` too. If the package itself is ESM then users will expect to be able to write `export default` in a `mocha.config.js`.

Thinking out loud: that _might_ be considered a breaking change. If someone happen to have a `module.exports.default = ...` in their `mocha.config.js` then that would change their behavior. [This Sourcegraph search for `default` in Mocha config files](https://sourcegraph.com/search?q=context:global+default+file:mocharc%5C..*+lang:JavaScript+&patternType=keyword&sm=0) shows that nobody in public seems to be doing this. So I think it's fine.

So, requesting changes: that the `.default ??` should always apply.